### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by jesperolsson-se**

### CODE REVIEW

**Review by npetzall — CHANGES_REQUESTED** [review]
ProgressListener is bad, most of the methods that "render"/write to console har only invoked if Html.
So ConsoleProgressListener doesn't do anything with the console except for 1 method that has something to do with exceptions.

Two things that could be done, 
1. skip the whole if htmlEnabled. It will almoste never be called if it's disabled. Has more with verbose than html, but only used when doing html. Strange stuff.

But the responsibilities could be divided. 
ConsoleProgressListener.java -> Tracked.java
Html -> Console.java

File: src/main/java/org/schemaspy/model/ConsoleProgressListener.java [hunk 1]
```diff
@@ -33,147 +31,76 @@
...
+    }
+
+    @Override
+    public long finished(Collection<Table> tables, Config config) {
+        finishedAt = System.currentTimeMillis();
+        return finishedAt - startedAt;
+    }
+
+    @Override
+    public String recoverableExceptionEncountered(String msg, Exception exc, String sql) {
```
Comment (npetzall) [inline src/main/java/org/schemaspy/model/ConsoleProgressListener.java:95]: Remove, this isn't progress, it not used, but declared by the interface.

File: src/main/java/org/schemaspy/model/ConsoleProgressListener.java [hunk 2]
```diff
@@ -33,147 +31,76 @@
...
+    private long startedGraphingDetailsAt;
+    private long finishedAt;
 
-	@Override
-	public void gatheringDetailsProgressed(Table table) {
-		if (render) {
-			System.out.print('.');
-		}
-	}
+    public ConsoleProgressListener() {
```
Comment (npetzall) [inline src/main/java/org/schemaspy/model/ConsoleProgressListener.java:42]: Doesn't do much with "Console", should Tracking, Timing, or Simple, Base

File: src/main/java/org/schemaspy/model/Html.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import org.schemaspy.Config;
+import org.schemaspy.cli.CommandLineArguments;
+
+import java.io.File;
+import java.util.Collection;
+
+/**
+ * Decorator for rendering console-based details for HTML.
+ */
+public class Html implements ProgressListener {
```
Comment (npetzall) [inline src/main/java/org/schemaspy/model/Html.java:12]: Most methods are used by Html generating Code. But what it does is write to System.out which in it self isn't great.

System.out is StdOut so may be StdOut or Console.

File: src/main/java/org/schemaspy/model/Html.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        long result = origin.finished(tables, config);
+        System.err.flush();
+        System.out.flush();
+        System.out.println("Wrote relationship details of " + tables.size() + " tables/views to directory '" + commandLineArguments.getOutputDirectory() + "' in " + result / 1000 + " seconds.");
+        System.out.println("View the results by opening " + new File(commandLineArguments.getOutputDirectory(), "index.html"));
+        return result;
+    }
+
+    @Override
+    public String recoverableExceptionEncountered(String msg, Exception exc, String sql) {
```
Comment (npetzall) [inline src/main/java/org/schemaspy/model/Html.java:105]: Should be removed from interface and thus from here.

**Review by jesperolsson-se — COMMENTED** [review]
The review comments are about pre-existing implementation. I'd be happy to resolve them in new PRs (after this one is merged). Do you have any concerns regarding the refactoring?

**Comment by npetzall** [comment]
The naming issue is introduced.

**Review by npetzall — APPROVED** [review]

**Comment by npetzall** [comment]
The removal is mentioned in #247

**Pull request merged**